### ISSUE TITLE
[rule-request]: Warn on empty lifecycle hooks

### ISSUE THREAD

**Issue by mgechev** [angular-eslint/angular-eslint#104]
Empty lifecycle hooks can have some performance implications without bringing any benefits. Since the runtime will process them we'll get:

- Slower processing time
- Larger memory consumption

```ts
@Component({... })
export class AppComponent implements OnInit {
  // This should warn
  ngOnInit() {}
}
```

**Comment by dev054** [comment]
Hey @mgechev ESLint already provides a rule that prevent [empty functions](https://eslint.org/docs/rules/no-empty-function). That said, do you think `angular-eslint` should still add a specific rule for lifecycle hooks?

**Comment by mgechev** [comment]
For lifecycle hooks this would be a bigger overhead compared to the standard scenario. I think it'll be valuable to expand the error message for Angular developers.
